functional-test/src/test/groovy/org/rundeck/util/common/execution/ExecutionUtils.groovy

```diff
@@ -0,0 +1,89 @@
+package org.rundeck.util.common.execution
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.rundeck.util.api.responses.execution.Execution
+import org.rundeck.util.api.responses.jobs.JobExecutionsResponse
+import org.rundeck.util.container.RdClient
+
+class ExecutionUtils {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
+    static final List<String> EXECUTION_FINISH_STATUSES = [
+            'aborted',
+            'failed',
+            'succeeded',
+            'timedout',
+            'other'].asImmutable()
+
+    static class Verifiers {
```

Comment:
I like this idea of a class with common Verifiers and Retrievers, but maybe they would be better as independent classes with many kinds of each as this api expands, instead of having different classes of them for each domain.
I.e: `Verifiers.executionFinished()` and `Verifiers.jobUpdated()` so they are easier to find by the uninitiated dev.
OR maybe: `Verifiers.Executions.executionFinished()` would be a good approach too.

Reply:
There are pros and cons to both approaches. Fortunately, such refactoring is easy with IDEs.
I'd propose we push in what we have and see how it picks up prior to premature refactoring.

Reply:
My only objective with this point it to prevent proliferation of classes when all could be managed by one or two, but as you comment we can improve that in a later step. 👍🏽